<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="OpenMendel : An Open Source Program for Statistical Genetics Analysis">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>OpenMendel</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/OpenMendel">View on GitHub</a>

          <h1 id="project_title">OpenMendel</h1>
          <h2 id="project_tagline">An Open Source Program for Statistical Genetics Analysis</h2>

        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h2>
<a id="openmendel-packages" class="anchor" href="#openmendel-packages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>OpenMendel Packages</h2>

<p>OpenMendel is an open source project implemented in the [Julia](http://julialang.org/) programming language that comprises a set of packages for statistical analysis to solve a variety of genetic problems. OpenMendel is the open source version of the [Mendel]((http://www.genetics.ucla.edu/software/mendel)) [Mendel](http://www.genetics.ucla.edu/software/mendel), a comprehensive package for exact statistical genetic analysis of qualitative and quantitative traits. Mendel offers 24 options for statistical genetic analysis. OpenMendel currently provides the analysis packages listed below, with additional options under development. We invite the genetics community to contribute to the OpenMendel Project.</p>

<h3>
<a id="packages" class="anchor" href="#packages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Packages:</h3>

<ul>
<li>
<a href="https://github.com/OpenMendel/MendelAimSelection.jl">AIM Selection</a> selects the SNPs that are most informative at predicting ancestry for your data â€” the best Ancestry Informative Markers (AIMs).</li>
<li>
<a href="https://github.com/OpenMendel/MendelEstimateFrequencies.jl">Estimate Frequencies</a> calculates a likelihood-based estimate of allele frequencies.</li>
<li>
<a href="https://github.com/OpenMendel/MendelGameteCompetition.jl">Gamete Competition analysis</a> implements a gamete competition analysis, which is a generalization of the TDT analysis.</li>
<li>
<a href="https://github.com/OpenMendel/MendelGeneDropping.jl">Gene Dropping</a> performs gene dropping with several useful options for the output.</li>
<li>
<a href="https://github.com/OpenMendel/MendelGeneticCounseling.jl">Genetic Counseling</a>  performs risk calculates for genetic counseling problems.</li>
<li>
<a href="https://github.com/OpenMendel/MendelGWAS.jl">GWAS</a> performs standard Genome-wide Association Studies.</li>
<li>
<a href="https://github.com/OpenMendel/MendelKinship.jl">Kinship</a> Kinship computes kinship and other identity coefficients.</li>
<li>
<a href="https://github.com/OpenMendel/MendelLocationScores.jl">Location Scores</a> maps a trait via the method of Location Scores, i.e., multipoint linkage analysis.</li>
<li>
<a href="https://github.com/OpenMendel/MendelTwoPointLinkage.jl">Two Point Linkage</a> performs two-point linkage analysis.</li>
</ul>

<h3>
<a id="utilities" class="anchor" href="#utilities" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Utilities</h3>

<p>OpenMendel also includes the following utility packages:</p>

<ul>
<li>
<a href="https://github.com/OpenMendel/Search.jl">Search</a> is a program for function optimization. Search permits bounds and linear constraints to be imposed on parameters and, in statistical applications, computes asymptotic standard errors and correlations of parameter estimates.</li>
<li>
<a href="https://github.com/OpenMendel/MendelBase.jl">MendelBase</a> contains the base functions of OpenMendel</li>
<li>
<a href="https://github.com/OpenMendel/SnpArrays.jl">SnpArrays</a> provides utilities for handling compressed storage of biallelic SNP data.</li>
<li>
<a href="https://github.com/OpenMendel/VarianceComponentModels.jl">Variance Components Models</a> provides utilities for fitting and testing variance components models, also known as linear mixed models.</li>
</ul>

<h3>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Installation</h3>

<p><em>Note: Three OpenMendel packages - <a href="https://github.com/OpenMendel/SnpArrays.jl">SnpArrays</a>, <a href="https://github.com/OpenMendel/Search.jl">Search</a>, and <a href="https://github.com/OpenMendel/MendelBase.jl">MendelBase</a> must be installed before any Mendel analysis packages will run.</em></p>

<p>Within Julia, use the package manager to install MendelPackageName:</p>

<pre><code>Pkg.clone("git@github.com:OpenMendel/MendelPackageName.jl.git")
</code></pre>

<p>This package supports Julia v0.4.</p>

<h3>
<a id="data-files" class="anchor" href="#data-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Data Files</h3>

<p>To run an analysis package you will need to prepare a Control file and have your data files available. The Control file holds the names of your data files and any optional parameters for the analysis. Details on the general format and contents of the Control and data files can be found on the MendelBase <a href="https://openmendel.github.io/MendelBase.jl">documentation page</a>. Descriptions of the specific options available within the any analysis package are in its documentation page.</p>

<p>There are example data files in the "docs" subfolder of each Mendel package, for example, ~/.julia/v0.4/MendelGWAS/docs.</p>

<h3>
<a id="running-the-analysis" class="anchor" href="#running-the-analysis" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Running the Analysis</h3>

<p>To run this analysis package, first launch Julia. Then load the package with the command:</p>

<pre><code> julia&gt; using MendelGeneDropping
</code></pre>

<p>Next, if necessary, change to the directory containing your files, for example,</p>

<pre><code> julia&gt; cd("~/path/to/data/files/")
</code></pre>

<p>Finally, to run the analysis using the parameters in the control file Control_file.txt use the command:</p>

<pre><code> julia&gt; GeneDropping("Control_file.txt")
</code></pre>

<p><em>Note: the package is called</em> MendelGeneDropping <em>but the analysis function is called simply</em> GeneDropping.</p>

<h3>
<a id="citation" class="anchor" href="#citation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Citation</h3>

<p>If you use this analysis package in your research, please cite the following reference in the resulting publications:</p>

<p><em>Lange K, Papp JC, Sinsheimer JS, Sripracha R, Zhou H, Sobel EM (2013) Mendel: The Swiss army knife of genetic analysis programs. Bioinformatics 29:1568-1570.</em></p>
      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
