<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="OpenMendel : An Open Source Program for Statistical Genetics Analysis">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>OpenMendel</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/OpenMendel">View on GitHub</a>

          <h1 id="project_title">OpenMendel</h1>
          <h2 id="project_tagline">An Open Source Program for Statistical Genetics Analysis</h2>

        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h2>
<a id="openmendel-project" class="anchor" href="#openmendel-project" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>OpenMendel Project</h2>

<p>OpenMendel is an open source project implemented in the <a href="http://julialang.org">Julia</a> programming language that comprises a set of packages for statistical analysis to solve a variety of genetic problems. OpenMendel is the open source version of <a href="http://www.genetics.ucla.edu/software/mendel">Mendel</a>, a comprehensive package for exact statistical genetic analysis of qualitative and quantitative traits. OpenMendel currently provides the analysis options listed below, with additional options under development. Each analysis option is a separate Julia package. We invite the genetics community to contribute to the OpenMendel Project.</p>

<h3>
<a id="analysis-options" class="anchor" href="#analysis-options" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Analysis Options:</h3>

<ul>
<li>
<a href="https://openmendel.github.io/MendelAimSelection.jl">AIM Selection</a> selects the SNPs that are most informative at predicting ancestry for your data â€” the best Ancestry Informative Markers (AIMs). [Package name: MendelAimSelection.]</li>
<li>
<a href="https://openmendel.github.io/MendelEstimateFrequencies.jl">Estimate Frequencies</a> calculates a likelihood-based estimate of allele frequencies. [Package name: MendelEstimateFrequencies.]</li>
<li>
<a href="https://openmendel.github.io/MendelGameteCompetition.jl">Gamete Competition</a> implements a gamete competition analysis, which is a generalization of the TDT analysis. [Package name: MendelGameteCompetition.]</li>
<li>
<a href="https://github.com/OpenMendel/MendelGeneDropping.jl">Gene Dropping</a> performs gene dropping with several useful options for the output. [Package name: MendelGeneDropping.]</li>
<li>
<a href="https://openmendel.github.io/MendelGeneticCounseling.jl">Genetic Counseling</a>  performs risk calculations for genetic counseling problems. [Package name: MendelGeneticCounseling.]</li>
<li>
<a href="https://github.com/OpenMendel/MendelGWAS.jl">GWAS</a> performs standard Genome-wide Association Studies. [Package name: MendelGWAS.]</li>
<li>
<a href="https://openmendel.github.io/MendelKinship.jl">Kinship</a> computes kinship and other identity coefficients. [Package name: MendelKinship.]</li>
<li>
<a href="https://github.com/OpenMendel/MendelLocationScores.jl">Location Scores</a> maps a trait via the method of Location Scores, i.e., multipoint linkage analysis. [Package name: MendelLocationScores.]</li>
<li>
<a href="https://openmendel.github.io/MendelTwoPointLinkage.jl">Two Point Linkage</a> performs two-point linkage analysis. [Package name: MendelTwoPointLinkage.]</li>
</ul>

<h3>
<a id="utilities" class="anchor" href="#utilities" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Utilities</h3>

<p>OpenMendel also includes the following utility packages:</p>

<ul>
<li>
<a href="https://openmendel.github.io/Search.jl">Search</a> is a program for function optimization. Search permits bounds and linear constraints to be imposed on parameters and, in statistical applications, computes asymptotic standard errors and correlations of parameter estimates.</li>
<li>
<a href="https://openmendel.github.io/MendelBase.jl">MendelBase</a> contains the base functions of OpenMendel</li>
<li>
<a href="https://openmendel.github.io/SnpArrays.jl/latest/">SnpArrays</a> provides utilities for handling compressed storage of biallelic SNP data.</li>
<li>
<a href="https://openmendel.github.io/VarianceComponentModels.jl/latest/">VarianceComponentModels</a> provides utilities for fitting and testing variance components models, also known as linear mixed models.</li>
</ul>

<h3>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Installation</h3>

<p><em>Note: Three OpenMendel packages - <a href="https://openmendel.github.io/SnpArrays.jl">SnpArrays</a>, <a href="https://openmendel.github.io/SnpArrays.jl/latest/">Search</a>, and <a href="https://openmendel.github.io/MendelBase.jl">MendelBase</a> must be installed before any Mendel analysis packages will run.</em></p>

<p>Within Julia, use the package manager to install the OpenMendel package, for example, MendelBase:</p>

<pre><code>Pkg.clone("git@github.com:OpenMendel/MendelBase.jl.git")
</code></pre>

<p>This package supports Julia v0.4.</p>

<h3>
<a id="data-files" class="anchor" href="#data-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Data Files</h3>

<p>To run an analysis package you will need to prepare a Control file and have your data files available. The Control file holds your data filenames and any optional parameters for the analysis. Details on the format and contents of the Control and data files can be found in the <a href="https://openmendel.github.io/MendelBase.jl">MendelBase documentation page</a>. Descriptions of the relevant parameters (<em>keywords</em>) available within each analysis package are in its documentation page (see the links above).</p>

<p>There are example data files in the "docs" subfolder of each Mendel package, for example, ~/.julia/v0.4/MendelGWAS/docs.</p>

<h3>
<a id="running-the-analysis" class="anchor" href="#running-the-analysis" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Running the Analysis</h3>

<p>To run an analysis package, first launch Julia. Then load the package, for example MendelGWAS, with the command:</p>

<pre><code> julia&gt; using MendelGWAS
</code></pre>

<p>Next, if necessary, change to the directory containing your files, for example,</p>

<pre><code> julia&gt; cd("~/path/to/data/files/")
</code></pre>

<p>Finally, to run the analysis using the parameters in your Control file, for example, Control_file.txt, use the command:</p>

<pre><code> julia&gt; GWAS("Control_file.txt")
</code></pre>

<p><em>Note: Mendel analysis packages are named with the Mendel prefix, for example,</em> MendelGWAS, <em>but the actual analysis functions do not have the prefix, for example,</em> GWAS.</p>

<h3>
<a id="citation" class="anchor" href="#citation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Citation</h3>

<p>If you use this analysis package in your research, please cite the following reference in resulting publications:</p>

<p><em>Lange K, Papp JC, Sinsheimer JS, Sripracha R, Zhou H, Sobel EM (2013). Mendel: The Swiss army knife of genetic analysis programs. Bioinformatics 29:1568-1570.</em></p>
      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
